### PR TITLE
feat: add fastify v4 codemods

### LINKED ISSUE #5574: creating codemods for fastify upgrade v4
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Building codemods for fastify upgrade v4.

At [Codemod](https://www.codemod.com), we are contributing open-source codemods for popular frameworks like [React](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#codemods), Remix, Nuxt, and others to help with their latest updates and changes. Right now, I'm working on the fastify v4 upgrade. I've gone through the upgrade guide and raised a [GitHub issue](https://github.com/codemod-com/codemod/issues/972) for the missing codemods. 

Looking forward to add these and help fastify users have a smooth migration experience.

### Motivation

The motivation to create codemods is that it would streamline the process for millions of users, and also ensure that the application stays efficient and secure while leveraging the latest advancements in Fastify.

### Example

These codemods would help migrate to fastify v4 real quick as running :

npx codemod fastify/4/migration-recipe which would include all the codemods.

### PR DESCRIPTION
**Pull request by arshcodemod**
closes #5574 creating codemods for fastify v4.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

But you need to fix the two CI jobs that are failing

**Comment by arshcodemod** [comment]
Thanks @mcollina , these CI jobs are fixed.

File: docs/Guides/Migration-Guide-V4.md [hunk 1]
```diff
@@ -6,6 +6,30 @@ Before migrating to v4, please ensure that you have fixed all deprecation
...
+- **fastify/4/remove-app-use**
+- **fastify/4/reply-raw-access**
+- **fastify/4/wrap-routes-plugin**
+- **fastify/4/await-register-calls**
+
+Each of these codemods automates the changes listed in the v4 migration guide.
+For a complete list of available Fastify codemods and further details, 
+see the [codemod registry](https://codemod.com/registry?q=fastify).
+
+
```
Comment (alexbit-codemod) [inline docs/Guides/Migration-Guide-V4.md:32]: You can add:

If you have any feedback for codemods run this command:
npx codemod feedback

**Comment by arshcodemod** [comment]
Thankyou! Onto v5 upgrade now :)

**Pull request merged**